src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp

```diff
@@ -159,6 +159,13 @@ std::vector<TimerInfo> GpuQueueSubmissionProcessor::ProcessGpuQueueSubmissionWit
   std::optional<GpuCommandBuffer> first_command_buffer =
       ExtractFirstCommandBuffer(gpu_queue_submission);
 
+  // The first command buffer acts as our reference needed to align GPU time based events in the
+  // CPU timeline. If we have no result for it -- which is the case if we started capturing within
+  // its execution -- we need to give up for this.
+  if (first_command_buffer.has_value() && first_command_buffer->begin_gpu_timestamp_ns() == 0) {
```

Comment:
What happens if `!first_command_buffer.has_value()`? Is that handled correctly in the calls that follow?

Reply:
Yes. The "first command buffer" is simply the first command buffer in this submission, if there are any. The code that follows always iterates of the command buffers, and does stuff for a concrete command buffer. Thus, in those cases the "first command buffer" is always set --  otherwise, we would just skip the loops.